### ISSUE TITLE
Feature: `connect` API sdk

### ISSUE THREAD

**Issue by smashah** [open-wa/wa-automate-nodejs#894]
Basically, it will make developing for remote instances as easy as local once made with `create`

- [x] Blocked by sockets implementation
- [x] Blocked by events updates
- [x] Requires a status endpoint

**Comment by smashah** [comment]
https://github.com/postmanlabs/postman-code-generators/tree/master

**Comment by smashah** [comment]
https://github.com/OpenAPITools/openapi-generator

**Comment by smashah** [comment]
https://github.com/swagger-api/swagger-codegen#to-generate-a-sample-client-library

**Comment by joedutra** [comment]
<img src="https://media.giphy.com/media/sNZddQMtrqpS0HEEMC/giphy.gif"></img>
Man, you know it sucks, how we copy your repo, if we have functions that you don’t even have, all for free, we perform almost every day, I came from Sulla and you too, but if you look at following the week, we redid it all lib again, now you will no longer be able to copy us, you copy and keep saying that we copy, nor compete with you ours and for free and your payment

**Comment by joedutra** [comment]
I come to apologize and say that we have nothing against this, only that they accuse us of copying and we never copied anything, if someone copied it was when someone changed our lib, I apologize and the intention was not to be slandered anymore of copying something we worked on every day. just like you, we will not do that anymore.

**Comment by Aminadav** [comment]
+1

**Comment by smashah** [comment]
blocked by #1100

**Comment by smashah** [comment]
@github-actions run

<details>
<summary>⚡ Release!  ⚡</summary>

```js
(async () => {
function exec(cmd) {
  console.log(execSync(cmd).toString());
}

// Config
const gitUserEmail = "github-actions[bot]@users.noreply.github.com";
const gitUserName = "github-actions[bot]";

exec(`echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > .npmrc`);
exec(`git config --global user.email "${gitUserEmail}"`);
exec(`git config --global user.name "${gitUserName}"`);
exec(`npm i -D`);
exec(`npm run release-ci minor`);

//comment on the issue
var result = execSync(`npx auto-changelog -o ./tempchangelog.txt --commit-limit false --template ./compact-keepachangelog.hbs --stdout`).toString();

    await postComment(result);

//create changelog image
exec(`npm run release-image`);
exec(`git commit -a -m 'updated release-image'`);
exec(`git push --force`);
  })();
```
</details>

**Comment by smashah** [comment]
# Changelog

## 🚀 Release [3.15.0](https://github.com/open-wa/wa-automate-nodejs/compare/3.14.1...3.15.0) (2021-05-17)

- 📝 docs: added `SocketClient` instructions and example #894 [`ef2437e`](https://github.com/open-wa/wa-automate-nodejs/commit/ef2437ead3491c8de3391b96edb7a3a2c9c8afde)
- Release 3.15.0 [`7def6fb`](https://github.com/open-wa/wa-automate-nodejs/commit/7def6fba2e0959e97739928354d8771b3b015272)
- ➕ deps: added `socket.io-client` dependency #894 [`7acbb94`](https://github.com/open-wa/wa-automate-nodejs/commit/7acbb946bdec8a841cb2c715f09c16e5bdf5fbbb)
- 🚨 road-2-esm: remove requires from `Client.ts` and fix eslint issues #1650 [`017d6fb`](https://github.com/open-wa/wa-automate-nodejs/commit/017d6fbe673c9fb7f211e461742037f0da0059b6)
- ✨ faat: `SocketClient.connect` to hook up to remote EASY API instances without needing webhooks #894 [`f544b93`](https://github.com/open-wa/wa-automate-nodejs/commit/f544b93c5619897acaa0d61869d8ca0f28098197)
- 🚨 road-2-esm: remove remaining `requires` from controllers #1650 [`ad34313`](https://github.com/open-wa/wa-automate-nodejs/commit/ad3431318f360ed6572e82650059ab66e05250ef)
- ✨ feat [CLI]: use `--socket` to expoe a socket.io middleware #1100 [`d263a37`](https://github.com/open-wa/wa-automate-nodejs/commit/d263a375f85ebe707c7c1a6e7fd10f5759ba0193)
- 🚨 road-2-esm: remove all `require`s and fix eslint issues in `browser.ts` #1650 [`9612ad5`](https://github.com/open-wa/wa-automate-nodejs/commit/9612ad56fc066a352897c58eaa00afdc1c1aa9a9)
- 🚨  road-2-esm: removed all `requires` and fixed eslint on initializer #1650 [`c40a400`](https://github.com/open-wa/wa-automate-nodejs/commit/c40a400d7514c3e52b4b587de90d0db24a06e12f)
- 🚨 road-2-esm: fix eslint issues in `events.ts` #1650 [`a0b79fc`](https://github.com/open-wa/wa-automate-nodejs/commit/a0b79fc77acecb52818de7018be144b0c9e9e10a)
- 🚨 road-2-esm: rollback migration from require to import for pptr-extra #1650 [`a24770f`](https://github.com/open-wa/wa-automate-nodejs/commit/a24770fa1612b1587dcc486d3578e6b71d53d1b4)
- 🐛  fix: export `SimpleListener` enum [`41ccf98`](https://github.com/open-wa/wa-automate-nodejs/commit/41ccf98a886e81958423eb0484129fb6e71c9bfc)

**Comment by smashah** [comment]
https://docs.openwa.dev/classes/connect_socket.socketclient.html
